### DISCUSSION TITLE
A new name for SixtyFPS

### DISCUSSION THREAD

**Discussion by ogoffart** [slint-ui/slint#636] [Ideas] [answered]
As explained in issue #635, we want to to change the name of SixtyFPS and we are open for suggestions

We need a name which we can derive
 - the rust crate name
 - the C++ namespaces
 - an extension for our language (which currently is `.60`)
 
Ideally the name can be reasonably short so the namespace is not too long (max 10 letters).
We would like to have something with is easy to spell correctly.
Not already taken: the crate name need to be available on crates.io. There shouldn't be another 
framework or software using that name.  Bonus if the .com is free, and the twitter handle, and the github account.

There is nothing to "win" and we might not even take any suggestions, but thanks 
for contributing 😀

**Comment by ghost** [comment]
**Ochre** a natural clay earth pigment, a mixture of _ferric_ oxide.
The spin might be too Rust specific though.
The crate has been taken, although the repo behind it hasn't had activity in 9 months.

**Reply by ogoffart** [reply]
Thanks for the proposal, that's a nice idea.

> The crate has been taken, although the repo behind it hasn't had activity in 9 months.

That's kind of a deal beaker. We do want the crate to be free.

**Reply by ghost** [reply]
Just fyi, If you ever need a crate that someone has, but has abandoned, you can contact them and ask them if they want to hand ownership to you via the dashboard at crates.io. Done it myself on an old crate I had, was quite easy.

**Reply by ogoffart** [reply]
good to know.
But in that case, it does not seem like it is a name that is parked. It has an actual repository and actual releases.

**Comment by dragotin** [comment]
**Lissous** : A short form of *Lissajous* (which is harder to spell). The inspiration comes from the magic Lissajous figures that are visually beautiful, sometimes dynamic, and with a nice and somehow simple mathematical and rational background. Just like your project :-)

The company name could be **Lissous**, the product could be combined with `Tk`for Toolkit.

On twitter (which is the hardest) **lissous** itself is taken, but lissoustk is still available (owned by a friend... ;-)

Actually... I like sixtyFPS :crying_cat_face:

**Reply by RiedleroD** [reply]
so… LTk?

**Comment by nicolasbauw** [comment]
I see sixtyfps as THE toolkit I've been waiting for so long.
So, I would suggest The Toolkit, crate name the-toolkit, .ttk as the language extension.
Sounds nice to me, although it is maybe a bit megalomaniac ?

**Reply by ogoffart** [reply]
nice one :-)
We are trying to concentrate on GUI and not be an universal toolkit like some other toolkit try to be. (do one thing but do it well) So maybe *theGUI* but that doesn't sound as nice

**Reply by oriontvv** [reply]
> I see sixtyfps as THE toolkit I've been waiting for so long. So, I would suggest The Toolkit, crate name the-toolkit, .ttk as the language extension. Sounds nice to me, although it is maybe a bit megalomaniac ?

It might be difficult to google information with such a name(The Toolkit)
What about Gui2LKIT, uitool, guidgets, izigui? not sure if some of them is owned

**Reply by nyanpasu64** [reply]
ttk is already the name of Themed Tk, so you should keep that in mind.

**Reply by RiedleroD** [reply]
maybe the GUI toolkit? TGTK? Although that might be too close to GTK…
The graphical interface? TGI?
The user interface kit? TUIK?

**Comment by DaMilyutin** [comment]
Keeping the word "Frames" with some adjective:
GFrames/GoodFrames/GUIFrames, QuickFrames, NiceFrames, BestFrames [per Second],
EasyFrames, FastFrames
?

I'd add many more when get inspired. :-)

**Comment by hunger** [comment]
Maybe Apelles? That's an ancient greek painter.

**Comment by prasanthu** [comment]
ogotron ;-)

**Reply by dfaure-kdab** [reply]
or troffart ? :-)

**Comment by bitbrat** [comment]
Liquid, LiquidUI

**Reply by tyranron** [reply]
Fluid = Fast Lightweight UI

**Reply by bjorn** [reply]
Fluit = Fast Lightweight (or Fluid) UI Toolkit - It also means "Whistle" in Dutch.

_(I've used the name for a visual UI tool I've started 6 years ago, but will probably never finish.)_

**Reply by CRogers** [reply]
I love Fluid - it's still in the spirit of the 60fps name for having smooth, fluid rendering.

If you want to make it more "Rusty" can call it [**Ferrofluid**](https://en.wikipedia.org/wiki/Ferrofluid).

**Comment by pieter-scholtz** [comment]
Iron oxide + sixtyFPS = Iron oxide hexahydrate 
; - )

**Comment by ghost** [comment]
Just a bit of fun:
So chrome is the widgets etc around the client area of a window.
So maybe a good name would be Coder-Chrome... :-P
I guess Kodac may take offence ;-)
(Showing my age here :-P )

**Comment by aetelani** [comment]
Mjölnir (mjolnir, ...) Tool for devs to iron out UIs with power of gods.

**Comment by bwalter** [comment]
A GUI (especially) for Rouille (= Rust) can only be named "Gouille". But be careful not to break the gouille(s)!

(of course just kidding ;))

**Comment by Congyuwang** [comment]
call it OneHundredAndTwentyHertz.

**Reply by hunger** [reply]
But my screen has 144Hz! Do I need two licenses then to cover all the frames? That do I do with the spare 96Hz? :-)

**Comment by ghost** [comment]
### Slant  - _Straightforward Lightweight and Native Toolkit_
_a particular point of view from which something is seen or presented.
"a new slant on science"_

**Your Mission Statement**


**L**ightweight
**S**traightforward
**N**ative

**Reply by tronical** [reply]
We like this suggestion a lot! We're thinking of going with a variation of it, because according to our cursory testing among native speakers it can have a slightly negative meaning.

**Slint** - **S**traightforward **Li**ghtweight **N**ative **T**oolkit

**Reply by raspher** [reply]
Well, there is a website called slant -> [slant.co](https://www.slant.co)

**Reply by ghost** [reply]
Congratz on the new name!

**Comment by jmaargh** [comment]
Please please please something that's relatively unique, relatively easy to search for (so not "frames", or "toolkit" or anything which will get mixed up in search results with generic things), and relatively easy to spell and say. Please don't fall into the trap of the `/e/` project (my personal award winner for "least-discoverable project name").

Off the top of my head, what about "lapis"? As in lapis lazuli the beautiful blue mineral. No crate name collisions, very few existing search results.

Or something like "doohick" (short for "doohicky", which is semi-synonymous with "widget"). Or "magig" (short for thingamagig", also a pun on "magic")

**Reply by araruna** [reply]
> Off the top of my head, what about "lapis"? As in lapis lazuli the beautiful blue mineral. No crate name collisions, very few existing search results.

Lapis is nice

**Reply by raspher** [reply]
Well, enemies are called GTK (GIMP Tool Kit) and Qt (which stands for... _**Cute**_?!). Short, memorable names are best bet.

You have my upvote, mainly because **blue color associates with sense of security** (Windows was not secure, but got a fortune - their choice was this color, a little beautified). But... Lapis is not rusty... You know, rusty color, a bit mix of red/orange 😄 

Let's shorten your suggestion:
- LTK... It will definitely lose the battle with Qt
- LUI (Lapis UI), or **LUIT** (Lapis UI Toolkit ; google translate pointed me here, it's finnish... and it translates to _read_)

In my opinion good direction, but maybe something more rusty, as in color?

**Comment by korintje** [comment]
# Didymium
- An old element name of a compound which mainly consists of **rusted** state of Neodymium, whose element number is **60**.
- The name is derived from Greek word: δίδυμο, which means **twin element** (like the crate that uses both logic and design-markup languages).
- Googlability is better than other common element names (like Hydrogen, Titanium, etc...) because it is deprecated element name.

**Reply by Acetinov** [reply]
I prefer shortly _**neodym**_, remaining the extension `.60` (atomic number of neodymium).

**Reply by pieter-scholtz** [reply]
There is also Neodymia , which is a synonym for Neodymium Oxide.

**Reply by Timmmm** [reply]
Too long and clunky. What's that toolkit? Neodymium something?

**Comment by teenjuna** [comment]
Maybe **_pictura_**? From [Wiktionary](https://en.wiktionary.org/wiki/pictura):

> The picture or image component of something, such as an emblem or poem, that contains a combination of imagery and text or symbols.

The crate name is free. I didn't find any mentions of frameworks with this name. The .com and Twitter/GH handles seems to be taken. Could be solved by adding _-ui_ or _-lib_: @picturalib
File extension: .ptra (or wait, is it too long?)

**Comment by teenjuna** [comment]
Another one from latin: **_visus_**
From [Wiktionary](https://en.wiktionary.org/wiki/visus#Latin):
> - The action of looking.
> - The power of sight.
> - A vision, sight, apparition.
> - Appearance

Crate is free.  .com/Twitter/Github are taken. Maybe @visuslib or @visusui?
File extension: .vis

**Comment by backtozero** [comment]
Slick
.slk

**Comment by hunger** [comment]
User r/hojjat12000 on reddit suggested: SixtyFPS UI Toolkit (Tools): SUIT?

I am posting that here so that the suggestion will be considered properly.

**Comment by FlorentBecker** [comment]
Alma, which is a simple, memorable and essentially unused name. It can express that the 60fps language lets you express the soul of your interface. It can also stand as an hommage to the Bauhaus artist [Alma Siedhoff-Buscher](https://en.wikipedia.org/wiki/Alma_Siedhoff-Buscher) for her work on building blocks (the _schiffbauspiele_) as well as giving objects (dolls) new usages through their interfaces (the _Wurfpuppen_).

**Comment by DaMilyutin** [comment]
guidance = gui + dance

**Comment by g-fb** [comment]
Rustik, rust + kit backwards.

**Comment by VasanthakumarV** [comment]
[Abalon](https://en.wikipedia.org/wiki/Abalone) - a mollusk which produces [mother of pearl](https://en.wikipedia.org/wiki/Nacre),
It is actually _abalone_, but it rhymes with baloney, hence dropping the _e_.
I am trying to draw a comparison between the decorative pearl and the beautiful Interfaces that sixtyfps helps build.

**Comment by andyfozzy** [comment]
Refresh
- Refers to the refresh rate of the monitor.
- Indicates a new way of building guis

**Comment by pickfire** [comment]
0fps, the UI toolkit so fast to not be bounded by fps

Stolen from zero cost abstraction

**Reply by teenjuna** [reply]
Seems like a bad advertisement 😅

**Reply by peter-scholtens** [reply]
On the other side of the number range is infinity. So call it **inftyFPS** and use the **∞** character for the file extension.
Or refer to the aim to include all OS and call it **gui4all**, or **omniUI**.

**Comment by nyanpasu64** [comment]
RtGui, because real-time guis should never drop frames (unlike browsers lol)... At least I hope this code can live up to that standard. My concern is it can get confused with RtAudio.

**Comment by birkett83** [comment]
You should call it grease.
It keeps the industrial theme of the name rust, but suggests smooth movement.

There is a crate called grease but it appears to be dead, so the author might be willing to give up the name.

**Comment by fdncred** [comment]
AnyFPS

I like the FPS terminology because it spells out exactly what your product is without having people think about the meaning of a clever name.

**Comment by Type1J** [comment]
UIbiquitous

**Comment by d4nshields** [comment]
What's the simplest name change that could possibly work?!  SixtyKFPS.  Leave the extension the same.  Less confusion during namespace migration.   Aim for the stars!

**Comment by mmjconolly** [comment]
BigNumberFPS

**Reply by RiedleroD** [reply]
Please, that's ridiculous.

I propose BigIntFPS.

**Comment by kisstank** [comment]
FluentFPS. Backwards and future proof, agnostic to whatever refreshrate is state of the art at a given point in time ;)

**Comment by aldebaranzbradaradjan** [comment]
What do you think of RQML ? Or RML (Rust Modeling Language), DML (Declarative Modeling Language) ?

**Comment by upyesp** [comment]
How about "odo", 
after the shape shifter.
Or, "guido".

**Reply by DaMilyutin** [reply]
guido - I like it. Sounds like Japanese budo, bushido or karate-do.
gui-do - A way of GUI.

**Reply by Timmmm** [reply]
Odo is great. Guido is just someone's name. Bit weird.

**Comment by WushuDan** [comment]
Illusory or Illusory Motion.

Apparent motion is the most common type of illusory motion and is perceived when images are displayed in succession at a specific frame rate such as in a movie.

**Comment by pmjobin** [comment]
Flare

1 : a sudden blaze of light the flare of a match.
2 : a blaze of light used to signal, light up something, or attract attention

Also, it rhymes with "flair", which implies good tastes or style.

**Comment by tronical** [comment]
Thanks for all the suggestions, keep them coming! We are reading them all :)

**Comment by pieter-scholtz** [comment]
With the word rustacean/crustaceans in mind , perhaps karkinos might not be a bad name , it means crab in greek.

**Reply by rrrodzilla** [reply]
Sesenta. Because even if you don't recognize the Spanish word for 60, it still sounds classy in any language to say Sesenta UI and there's a lot of room to explore there with a logo. Easy to say, easy to spell. In addition, the name Sesenta serves as both a nod to the historical roots of the crate but also allows you to keep the .60 file extension. The crate name, Twitter handle and a variety of domain extensions (such as sesenta.io) are available.

**Comment by jrmoulton** [comment]
guiframe - .com available, crate available
fastgui/ fastui - .com taken, crate available
hyperui - .com taken, crate available
hyperframe - .com taken, crate available

**Reply by DaMilyutin** [reply]
rustgui
rustygui
😄

**Comment by diamant3** [comment]
corrodeGUI or corrodeUI? 😄

**Reply by diamant3** [reply]
graphics + pixel = grapix or graphix? 😅

**Comment by shaoron** [comment]
Euclase
Short name, nice blue gem. 
.com domain for sale

**Comment by DaMilyutin** [comment]
Me playing words again...

GUISE
(seems funny to me to associate this word with GOOSE and make goose as mascot 🤣)

noun
- general external appearance; aspect; semblance:  (perfect match for GUI meaning and for sound; extension .gui is awesome for this)
- assumed appearance or mere semblance.
- style of dress.                                                          (GUI has a lot of styles, too)
- Archaic. manner; mode.
verb (used with object), guised, guis·ing.
- to dress.
verb (used without object), guised, guis·ing.
Scot. and North England. to appear or go in disguise.

Also,
GREASE - makes everithing go smooth and fluent;
GEARS - what things are made of;
GUIRS - some GUI in GEARS.
Still GUISE is my fav.

**Comment by snnsnn** [comment]
Hi, i can donate the crate oro, sky, can, or ivy. All short and easy to remember. Oro means gold in spanish.

**Reply by hunger** [reply]
Wow, thanks for that gracious offer!

**Comment by pieter-scholtz** [comment]
**Some ideas :** 

Pochade , from what I understand means to rapidly sketch/paint. https://www.youtube.com/watch?v=OHMCh6aEaLU

Some Xhosa words :
painter = umzobi
beautiful = entle
draw = zoba

Some Zulu words:
painter = umdwebi
beautiful = Enhle
draw = dweba

**Comment by RiedleroD** [comment]
Prima? From Latin: patria → native, rubigo → Rust, imagines → images
Maybe PrimaTk.

Neither is taken on crates.io, the former is taken on Twitter, and primatk.com is open for sale.

**Comment by lukas-jung** [comment]
what about xty?
it is a substring of sixtyfps and the x kind of suggests a variable which makes sense for a modern variable framerate UI framework, which sixtyfps eventually will become, i guess.

Then the file extension would be .x0 and would be pronounced x-tea.

the crate is available as well.

**Reply by DaMilyutin** [reply]
Fancy name. I like it.
Why extension .x0 but not .xty or .xt ?

**Comment by hunger** [comment, marked as answer]
We have a new name! We are doing work to adopt it and have a new release under a new name soon:-)
